tests/quinn.rs

```diff
@@ -61,10 +75,19 @@ fn configure_server() -> anyhow::Result<(ServerConfig, Vec<u8>)> {
     let cert = rcgen::generate_simple_self_signed(vec!["localhost".into()])?;
     let cert_der = cert.serialize_der()?;
     let priv_key = cert.serialize_private_key_der();
-    let priv_key = rustls::PrivateKey(priv_key);
-    let cert_chain = vec![rustls::Certificate(cert_der.clone())];
+    let priv_key = rustls::pki_types::PrivatePkcs8KeyDer::from(priv_key);
+    let cert_chain = vec![rustls::pki_types::CertificateDer::from(cert_der.clone())];
+
+    let crypto_server_config = rustls::ServerConfig::builder_with_provider(Arc::new(
+        rustls::crypto::ring::default_provider(),
+    ))
+    .with_protocol_versions(&[&rustls::version::TLS13])
+    .expect("valid versions")
+    .with_no_client_auth()
+    .with_single_cert(cert_chain, priv_key.into())?;
+    let quic_server_config = QuicServerConfig::try_from(crypto_server_config)?;
+    let mut server_config = ServerConfig::with_crypto(Arc::new(quic_server_config));
```

Comment:
Sad how much more verbose the new rustls code is

Reply:
yeah, this took me a while to get my head around.  The reason it is more verbose for us is that we want to use the `.builder_with_provider()` version of things, rather than rely on per-process defaults.  If we did the latter we'd still get all the shortcuts provided by Quinn.

Personally I don't really like per-process globals like this, seems like @dignifiedquire also didn't because I followed the style he started.  Maybe there's room here for us to contribute better Quinn support for this so that this stuff becomes equally easy?